### ISSUE TITLE
PathSubtree takes priority over Path with wildcard

### ISSUE THREAD

**Issue by NorthFury** [zalando/skipper#1018]
**Describe the bug**
Even though a route has more predicates, the `PathSubtree` predicates trumps the `Path` predicate.

**To Reproduce**
```
route1: Path("/collections/**")
&& PathRegexp("collections/(one|two)")
-> status(200)
-> <shunt>;

route2: PathSubtree("/collections/")
-> status(201)
-> <shunt>;
```
Request for `"/collections/one"`
**Expected behavior**
it should return 200
**Observed behavior**
it returns 201

it seems to be related to #375

**Comment by aryszka** [comment]
thanks for reporting this bug

**Comment by aryszka** [comment]
fixed in #1200
